### PR TITLE
CLOUDP-167770: migrate 'alerts' CLI stores to v2 sdk

### PR DESCRIPTION
**Pull request by matt-condon**
## Proposed changes

_Jira ticket:_ CLOUDP-167770

Migration to sdkv2 of AlertDescriber, AlertLister, AlertAcknowledger

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .DS_Store
Comment (fmenezes) [inline .DS_Store:1]: maybe delete this file?

File: internal/cli/atlas/alerts/list.go [hunk 1]
```diff
@@ -59,15 +57,6 @@ func (opts *ListOpts) Run() error {
 	return opts.Print(r)
 }
 
-func (opts *ListOpts) newAlertsListOptions() *atlas.AlertsListOptions {
-	o := &atlas.AlertsListOptions{
-		Status:      opts.status,
-		ListOptions: *opts.NewListOptions(),
```
Comment (fmenezes) [inline internal/cli/atlas/alerts/list.go:65]: have we stopped paginating?

**Review by gssbzn — APPROVED** [review]
LGTM thanks

**Review by fmenezes — CHANGES_REQUESTED** [review]
I see flags for pageNum and no data being passed to the store, can you confirm if we are indeed paginating or not for the list command?

**Comment by matt-condon** [comment]
> I see flags for pageNum and no data being passed to the store, can you confirm if we are indeed paginating or not for the list command?

Thanks for flagging @fmenezes 
I amended the approach for this PR to use structs, and included the listOptions

File: internal/cli/atlas/alerts/list_test.go [hunk 2]
```diff
@@ -23,24 +23,25 @@ import (
...
+	expected := &admin.PaginatedAlert{
 		Links: nil,
-		Results: []mongodbatlas.Alert{
+		Results: []admin.AlertViewForNdsGroup{
 			{
-				ID:            "test",
-				EventTypeName: "test",
-				Status:        "test",
-				MetricName:    "test"},
+				Id:            pointer.Get("test"),
```
Comment (wtrocki) [inline internal/cli/atlas/alerts/list_test.go:39]: [Unrelated] We need to document SDK methods that create integer or string pointers so users can use SDK without having to rely on pointer lib etc.

**Review by wtrocki — APPROVED** [review]

File: internal/cli/atlas/alerts/list_test.go (hunk 2 above)
Comment (wtrocki) [reply internal/cli/atlas/alerts/list_test.go:39]: creating jira

**Review by wtrocki — APPROVED** [review]

**Review by fmenezes — APPROVED** [review]

**Review by swattyT — APPROVED** [review]
LGTM!

**Pull request merged**